roxie/ccd/ccdserver.cpp

```diff
@@ -21486,7 +21486,7 @@ class CRoxieServerDiskReadActivityFactory : public CRoxieServerActivityFactory
         {
             bool isOpt = (helper->getFlags() & TDRoptional) != 0;
             OwnedRoxieString fileName(helper->getFileName());
-            datafile.setown(_queryFactory.queryPackage().lookupFileName(fileName, isOpt, true, true, _queryFactory.queryWorkUnit()));
+            datafile.setown(_queryFactory.queryPackage().lookupFileName(fileName, isOpt, true, true, _queryFactory.queryWorkUnit(), !queryFactory.isDynamic()));
```

Comment:
Same considerations as in ccdactivities - most of the calls in this function are already inside checks that include this test (so we should pass true). The others probably SHOULD be inside tests that include seeing if the workunit is dynamic...